pkg/services/github.go

```diff
@@ -598,3 +702,17 @@ func (g gitHubService) Send(notification Notification, _ Destination) error {
 
 	return nil
 }
+
+// PullRequestsServiceAdapter adapts GitHub's PullRequestsService to our interface
+type pullRequestsServiceAdapter struct {
+	service *github.PullRequestsService
+}
+
+func (a *pullRequestsServiceAdapter) ListPullRequestsWithCommit(ctx context.Context, owner string, repo string, sha string, opts *github.PullRequestListOptions) ([]*github.PullRequest, *github.Response, error) {
+	// Convert PullRequestListOptions to ListOptions
+	listOpts := &github.ListOptions{
+		Page:    opts.Page,
+		PerPage: opts.PerPage,
+	}
```

Comment:
I'm seeing this panic in our cluster, it seems like we're not handling `opts` being `nil` 🤔